deploy-service/common/src/main/java/com/pinterest/deployservice/bean/EnvironBean.java

```diff
@@ -228,10 +231,35 @@ public void validate() throws IllegalArgumentException {
         if (this.chatroom != null && !this.chatroom.matches(chatRegex)) {
             throw new IllegalArgumentException(String.format("Chatroom must match regex %s", chatRegex));
         }
+    }
+
+    public void stageTypeValidate(EnvironBean origBean) throws Exception {
```

Comment:
i wonder if it might be better to move this check with "origBean" out to the envStages.java file as it is a very specific business logic. leaving "this.stage_type == EnvType.DEFAULT" check here is fine as it is part of the standard data model validation,i.e, no origbean info is needed.

Reply:
I updated it so that all the business logic is in EnvStages